
## Analysis


Analysis was performed using R:

| System       | Info
|:-------------|-----------------------------------|
|R version     | 3.3.2 (2016-10-31)                |
|Platform      | x86_64-apple-darwin13.4.0 (64-bit)|
|Running under | OS X El Capitan 10.11.6           |
|locale        | en_GB.UTF-8                       |



| Base package | Version | Author             | Non-base package | Version    | Author                            |
|--------------|---------|--------------------|------------------|------------|-----------------------------------|
| graphics     | 3.3.2   | R Core Team (2016) | seqinr           | 3.3-3      | Charif, D and Lobry, JR (2007)    |
| grDevices    | 3.3.2   | R Core Team (2016) | plyr             | 1.8.4      | Wickham, H (2011)                 |
| utils        | 3.3.2   | R Core Team (2016) | primerTree       | 1.0.3      | Hester, J (2013)                  |
| datasets     | 3.3.2   | R Core Team (2016) | gridExtra        | 2.2.1      | Auguie, B (2016)                  |
| methods      | 3.3.2   | R Core Team (2016) | directlabels     | 2015.12.16 | Hocking, TD (2015)                |
| base         | NA      | R Core Team (2016) | optparse         | 1.3.2      | Davis, TL (2015)                  |
| stats        | 3.3.2   | R Core Team (2016) | -                | -          | -                                 |


### Running analysis from the command line

1. download R scripts
  - [primer_blast_v1.R](primer_blast_v1.R)
  - [primer_blast_mismatch_plots_v1.R](primer_blast_mismatch_plots_v1.R)
2. download input file for primer_blast_v1.R
  - [primer_blast_inputs.csv](primer_blast_inputs.csv)
3. input file for primer_blast_mismatch_plots_v1.R
  - primer_blast_outputs.csv  
    *(file generated by primer_blast_v1.R)*
4. open a terminal session


```
## Replace "PATH"

## Submit primer pairs to NCBI Primer-Blast

Rscript PATH/primer_blast_v1.R -i PATH/primer_blast_inputs.csv -o PATH/primer_blast_outputs.csv


## Reproduce Figure 1 from the paper

Rscript PATH/primer_blast_mismatch_plots_v1.R -i PATH/primer_blast_outputs.csv -o PATH/primer_blast_mismatches
```

Output will be written out to the PATH specified, *overwriting any existing file
of the same name*.



### Running analysis from a interactive R session

1. download R scripts
  - [primer_blast_interactive_v1.R](primer_blast_interactive_v1.R])
  - [primer_blast_mismatch_plots_interactive_v1.R](primer_blast_mismatch_plots_interactive_v1.R)
2. download input file for primer_blast_interactive_v1.R
  - [primer_blast_inputs.csv](primer_blast_inputs.csv)
3. input file for primer_blast_mismatch_plots_interactive_v1.R
  - primer_blast_outputs.csv  
  *(file generated by primer_blast_interactive_v1.R)*
4. open a terminal session an interactive R session


```
## Replace "PATH"
## Scripts can be called using source(), else by copying an pasting into the interface.

## Submit primer pairs to NCBI Primer-Blast

infl <- "PATH/primer_blast_inputs.csv"
outfl <- "PATH/primer_blast_outputs.csv"

source("PATH/primer_blast_interactive_v1.R")


## Reproduce Figure 1 from the paper

infl <- "PATH/primer_blast_outputs.csv"
outfl <- "PATH/primer_blast_mismatches"

source("PATH/primer_blast_mismatch_plots_interactive_v1.R")
```

Output will be written out to the PATH specified, *overwriting any existing file
of the same name*.



## References

#### Software and applications

Auguie, B. (2016). gridExtra: Miscellaneous Functions for "Grid" Graphics. R package version 2.2.1. https://CRAN.R-project.org/package=gridExtra

Charif, D and Lobry, JR. (2007). SeqinR 1.0-2: a contributed package to the R project for statistical computing devoted to biological sequences retrieval and analysis. In: Bastolla, U, Porto, M, Roman, HE and Vendruscolo, M ed. Structural approaches to sequence evolution: Molecules, networks, populations. New York, Springer Verlag, pp 207-232. [ISBN : 978-3-540-35305-8]

Davis, TL. (2015). optparse: Command Line Option Parser. R package version 1.3.2. https://CRAN.R-project.org/package=optparse

Hester, J. (2013). primerTree: Visually Assessing the Specificity and Informativeness of Primer Pairs. R package version 1.0.3. https://CRAN.R-project.org/package=primerTree  [https://github.com/jimhester/primerTree]

Hocking, TD. (2015). directlabels: Direct Labels for Multicolor Plots. R package version 2015.12.16. https://CRAN.R-project.org/package=directlabels

NCBI Resource Coordinators. (2016). Database resources of the National Center for Biotechnology Information. Nucleic Acids Research, 44(Database issue), D7–D19. http://doi.org/10.1093/nar/gkv1290

O’Leary, NA, Wright, MW, Brister, JR, Ciufo, S, Haddad, D, McVeigh, R, … Pruitt, KD. (2016). Reference sequence (RefSeq) database at NCBI: current status, taxonomic expansion, and functional annotation. Nucleic Acids Research, 44(Database issue), D733–D745. http://doi.org/10.1093/nar/gkv1189

R Core Team. (2016). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. URL
https://www.R-project.org/

Sievers F, Wilm A, Dineen DG, Gibson TJ, Karplus K, Li W, Lopez R, McWilliam H, Remmert M, Söding J, Thompson JD, Higgins DG. (2011). Fast, scalable generation of high-quality protein multiple sequence alignments using Clustal Omega. Molecular Systems Biology 7:539 DOI:10.1038/msb.2011.75
http://www.clustal.org/omega/

Stewart Heitmann. argtable-2.13.x2. An ANSI C library for parsing GNU style command line arguments
http://argtable.sourceforge.net/

Wickham, H. (2011). The Split-Apply-Combine Strategy for Data Analysis. Journal of Statistical Software, 40(1), 1-29.
http://www.jstatsoft.org/v40/i01/

Ye, J, Coulouris, G, Zaretskaya, I, Cutcutache, I, Rozen, S, & Madden, TL. (2012). Primer-BLAST: A tool to design target-specific primers for polymerase chain reaction. BMC Bioinformatics, 13, 134. http://doi.org/10.1186/1471-2105-13-134
