
## Data

Analysis was performed on nine universal bacterial 16S ribsomal primer sets (Table 1., [primer_blast_inputs.csv](primer_blast_inputs.csv)),
using NCBI Primer-Blast with Fungi RefSeq database (NCBI Resource Coordinators. (2016); Ye, et al. (2012); O’Leary, et al. (2016)).


**Table 1.** Source of universal bacterial primer pairs used in the paper.

| Source                                 | FRWD  | Sequence             | REV   | Sequence                 |
|----------------------------------------|-------|----------------------|-------|--------------------------|
| Caporaso et al. (2011)*                 | 515F  | GTGYCAGCMGCCGCGGTAA  | 806R  | GGACTACNVGGGTWTCTAAT     |
| Fierer et al. (2005)                   | 338F  | ACTCCTACGGGAGGCAGCAG | 518R  | ATTACCGCGGCTGCTGG        |
| Heuer et al. (1999)                    | 968F  | AACGCGAAGAACCTTACC   | 1378R | CGGTGTGTACAAGGCCCGGGAACG |
| Human Microbiome Project (HMP) (2012)  | 357F  | CCTACGGGAGGCAGCAG    | 926R  | CCGTCAATTCMTTTRAGT       |
| Maeda et al. (2003)                    | 1048F | GTGSTGCAYGGYTGTCGTCA | 1175R | ACGTCRTCCMCACCTTCCTC     |
| Muyzer et al. (1993)                   | 357F  | CCTACGGGAGGCAGCAG    | 518R  | ATTACCGCGGCTGCTGG        |
| Schwieger and Tebbe (1998)             | 519F  | CAGCAGCCGCGGTAATAC   | 907R  | CCGTCAATTCCTTTGAGTTT     |
| Shakya et al. (2013)                   | 27F   | AGAGTTTGATYMTGGCTCAG | 534R  | TYACCGCGGCTGCTGG         |
| Weisburg et al. (1991)                 | 27F   | AGAGTTTGATCCTGGCTCAG | 1492R | TACCTTGTTACGACTT         |


Where: * Sequences used are as per those in the Earth Microbiome Project (Gilbert et al., 2014) 16S Illumina Amplicon protocol.



## References
#### Primer sets and database resources

Caporaso, JG, Lauber, CL, Walters, WA, Berg-Lyons, D, Lozupone, CA, Turnbaugh, PJ, Fierer, N and Knight,R. (2011). Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample. Proceedings of the National Academy of Sciences of the United States of America, 108(Suppl 1), 4516–4522. Epub 2010 Jun 3. http://doi.org/10.1073/pnas.1000080107
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3063599/bin/supp_108_suppl.1_4516__index.html

Fierer, N, Jackson, JA, Vilgalys, R and Jackson, RB. (2005). Assessment of Soil Microbial Community Structure by Use of Taxon-Specific Quantitative PCR Assays. Applied and Environmental Microbiology, 71(7), 4117–4120. http://doi.org/10.1128/AEM.71.7.4117-4120.2005

Gevers D, Knight R, Petrosino JF, Huang K, McGuire AL, Birren BW, … Huttenhower, C. (2012). The Human Microbiome Project: A Community Resource for the Healthy Human Microbiome. PLoS Biology, 10(8), e1001377. http://doi.org/10.1371/journal.pbio.1001377

Gilbert, JA, Jansson, JK, and Knight, R. (2014). The Earth Microbiome project: successes and aspirations. BMC Biology, 12(1). http://doi.org/10.1186/s12915-014-0069-1
http://www.earthmicrobiome.org/protocols-and-standards/16s/

Heuer, H, Hartung, K, Wieland, G, Kramer, I and Smalla, K. (1999). Polynucleotide probes that target a hypervariable region of 16S rRNA genes to identify bacterial isolates corresponding to bands of community fingerprints. Applied and Environmental Microbiology 65(3), pp.1045–1049.

Maeda, H, Fujimoto, C, Haruki, Y, Maeda, T, Kokeguchi, S, Petelin, M, Arai, H, Tanimoto, I, Nishimura, F and Takashiba, S. (2003). Quantitative real-time PCR using TaqMan and SYBR Green for *Actinobacillus actinomycetemcomitans*, *Porphyromonas gingivalis*, *Prevotella intermedia*, tetQ gene and total bacteria. FEMS Immunology and Medical Microbiology 39(1), pp.81-86. http://dx.doi.org/10.1016/S0928-8244(03)00224-4

Muyzer, G, De Waal, EC and Uitterlinden, AG. (1993). Profiling of complex microbial populations by denaturing gradient gel electrophoresis analysis of polymerase chain reaction-amplified genes coding for 16S rRNA. Applied and Environmental Microbiology 59(3), pp.695–700.

NCBI Resource Coordinators. (2016). Database resources of the National Center for Biotechnology Information. Nucleic Acids Research, 44(Database issue), D7–D19. http://doi.org/10.1093/nar/gkv1290

O’Leary, NA, Wright, MW, Brister, JR, Ciufo, S, Haddad, D, McVeigh, R, … Pruitt, KD. (2016). Reference sequence (RefSeq) database at NCBI: current status, taxonomic expansion, and functional annotation. Nucleic Acids Research, 44(Database issue), D733–D745. http://doi.org/10.1093/nar/gkv1189

Shakya, M, Quince, C, Campbell, JH, Yang, ZK, Schadt, CW and Podar, M. (2013). Comparative metagenomic and rRNA microbial diversity characterization using Archaeal and Bacterial synthetic communities. Environmental Microbiology, 15(6), 1882–1899. http://doi.org/10.1111/1462-2920.12086

Schwieger F and Tebbe, CC. (1998). A new approach to utilize PCR-single-strand-conformation polymorphism for 16S rRNA gene-based microbial community analysis. Applied and Environmental Microbiology, 64(12), 4870–4876.

Weisburg WG, Barns SM, Pelletier DA, Lane DJ (1991). 16S ribosomal DNA amplification for phylogenetic study. Journal of Bacteriology, 173(2), 697–703.

Ye, J, Coulouris, G, Zaretskaya, I, Cutcutache, I, Rozen, S, & Madden, TL. (2012). Primer-BLAST: A tool to design target-specific primers for polymerase chain reaction. BMC Bioinformatics, 13, 134. http://doi.org/10.1186/1471-2105-13-134
